feat: update redirection docs for pr12350 on nest (#2846)
nestjs/docs.nestjs.com: content/controllers.md

@Redirect('https://nestjs.com', 301)
```

> > info **Hint** Sometimes you may want to determine the HTTP status code or the redirect URL dynamically. Do this by returning a `HttpRedirectResponse` imported from `@nestjs/common`.

Reviewer: ```suggestion
> info **Hint** Sometimes you may want to determine the HTTP status code or the redirect URL dynamically. Do this by returning an object following the `HttpRedirectResponse` interface (from `@nestjs/common`).
```
